docker-compose/test.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# start docker daemon
+source /docker-lib.sh
+start_docker
+
+echo ${DOCKER_PASSWORD} | docker login -u ${DOCKER_USERNAME} --password-stdin
+unset DOCKER_USERNAME
+unset DOCKER_PASSWORD
+
+sanitise_image_name() {
+  echo ${1//[^a-zA-Z0-9_-]/-}
+}
+
+import_image() {
+  local image_name="$1"
+  local sanitised_image_name=$(sanitise_image_name "${image_name}")
+  docker load < "$CONCOURSE_WORKDIR/image-cache/${sanitised_image_name}"
+}
+
+set -e
+ARGV_DIRECTORY="$1"
+set -u
+
+[[ "${DEBUG}" == "false" ]] || set -x
+
+HERE="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+CONCOURSE_WORKDIR=$(pwd)
+
+pushd $ARGV_DIRECTORY
+
+[[ -f docker-compose.test.yml ]] || exit 0
+
+base_org=$(cat .git/.version | jq -r '.base_org')
+base_repo=$(cat .git/.version | jq -r '.base_repo')
+
+# Read the details of what we should build from .resinci.yml
+builds=$(${HERE}/../shared/resinci-read.sh \
+  -b $(pwd) \
+  -l docker \
+  -p builds | jq -c '.[]')
+
+if [ -n "$builds" ]; then
+  for build in ${builds}; do
+    repo=$((echo ${build} | jq -r '.docker_repo') || \
+      (echo "${base_org}/${base_repo}"))
+    import_image "$repo"
+  done
+else
+  import_image "${base_org}/${base_repo}"
+fi
+
+sut=$(yq read repo.yml 'sut')
+
+if [ "${sut}" == "null" ]; then
```

Comment:
I don't understand this comparison... If `repo.yml` does not contain the `sut` key or it has no value, the `sut` variable will be blank (as there's no `null` concept in bash) and this check will always fail... am I missing something?

Reply:
`yq` sets it to `null` if not found (the string `null` :P)

Reply:
We discussed this via Flowdock. Apparently the `yq` version @nazrhom is using is the same as the CI (v2.3) and returns the `null` string if the key does not exist. Newer `yq` versions return blank (tested in my machine, v3.1.2). We keep it like this until we update `yq`